application.nix

```diff
@@ -126,6 +123,28 @@ rec {
         wantedBy = [ "multi-user.target" ];
       };
 
+      # Monitor hotplugging
+      services.udev.extraRules = ''
+        ACTION=="change", SUBSYSTEM=="drm", RUN+="${pkgs.systemd}/bin/systemctl restart select-display.service"
+      '';
+      systemd.services."select-display" = {
+        description = "Select best display to output to";
+        serviceConfig.Type = "oneshot";
+        serviceConfig.ExecStart = "${pkgs.bash}/bin/bash ${./application/select-display.sh}";
+        serviceConfig.User = "play";
+        serviceConfig.After = "multi-user.target";
```

Comment:
I would expect this to be linked (through some rule) to the `graphical.target` instead? Also, I don't think the `After=`  will actually wait for the `multi-user.target`, so the `udev` rule will launch this too  early?

Reply:
Honestly I'd need a moment to refresh my memory on the targets. I dimly remember trying to rely on a target that I then realized would not be present on our minimal system. But I just checked for `graphical.target` inside a VM PlayOS and it's there, so plausibly yes.

Regarding the ordering conditions, I understand `After=` to actually wait:

> After= ensures [...] that the listed unit is fully started up before the configured unit is started.

https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#Before=

Reply:
> Regarding the ordering conditions, I understand After= to actually wait <..>

Yes, but only if both are being started at the same time, i.e. it is a "soft" dependency, from the same man page:

> If unit foo.service contains the setting Before=bar.service and **both units are being started**, bar.service's start-up is delayed until foo.service has finished starting up. After= is the inverse of Before=,

Reply:
I see how you read it, though I wouldn't have read it like that.

What if we use both `After=` and `Requires=`?